dpnp/dpnp_algo/dpnp_arraycreation.py

```diff
@@ -0,0 +1,275 @@
+import operator
+
+import numpy
+
+import dpnp
+import dpnp.dpnp_container as dpnp_container
+import dpnp.dpnp_utils as utils
+
+__all__ = [
+    "dpnp_geomspace",
+    "dpnp_linspace",
+    "dpnp_logspace",
+]
+
+
+def _get_sycl_queue(sycl_queue_alloc, sycl_queue, device):
+    if sycl_queue is None and device is None:
+        sycl_queue = sycl_queue_alloc
+    return dpnp.get_normalized_queue_device(
+        sycl_queue=sycl_queue, device=device
+    )
+
+
+def _get_temporary_usm_type(usm_type_alloc, usm_type):
+    if usm_type is None:
+        usm_type = "device" if usm_type_alloc is None else usm_type_alloc
+    return usm_type
+
+
+def _list_to_array(a, usm_type, sycl_queue, scalar=False):
+    if not hasattr(a, "dtype") and (scalar or not dpnp.isscalar(a)):
+        return dpnp.asarray(a, usm_type=usm_type, sycl_queue=sycl_queue)
+    return a
+
+
+def _copy_by_usm_type(a, dtype, usm_type, tmp_usm_type, sycl_queue_normalized):
+    if usm_type is None:
+        usm_type = tmp_usm_type
+        if not hasattr(a, "usm_type"):
+            res = dpnp.asarray(
+                a,
+                dtype=dtype,
+                usm_type=usm_type,
+                sycl_queue=sycl_queue_normalized,
+            )
+        else:
+            res = dpnp.asarray(a, dtype=dtype, sycl_queue=sycl_queue_normalized)
+    else:
+        res = dpnp.asarray(
+            a, dtype=dtype, usm_type=usm_type, sycl_queue=sycl_queue_normalized
+        )
+    return res

```

Comment:
looks the same as:
```suggestion
    if usm_type is None and not hasattr(a, "usm_type"):
        usm_type = tmp_usm_type

    return dpnp.asarray(a, dtype=dtype, usm_type=usm_type, sycl_queue=sycl_queue_normalized)
```
because if `usm_type is None`, then dpnp.asarray(...,) is the same as dpnp.asarray(..., usm_type=None)